Padatious intents

Tony763 opened the pull request:
- Rewrote all intents to padatious
- Fix in automation intent
- Address PR #38

## Description:

## Checklist:
  - [ ] Code is Python 3.x compatible
  - [ ] Travis is passing tests **Your PR cannot be merged unless tests pass**
  - [ ] No PEP Bot Errors or Issues

Tony763 committed:
Logger

Removed imported logger
Pointed all debug messages to self.log
Address PR #38

Tony763 committed:
Padatious intents

- Rewrote all intents to padatious
- Fix in automation intent
- Adress PR #38

Tony763 committed:
Padatious intents

- Rewrote all intents to padatious
- Fix in automation intent
- Adress PR #38

stratus-ss commented on __init__.py:
```diff
@@ -192,6 +177,31 @@ def handle_light_decrease_intent(self, message):
 
+    @intent_handler('automation.intent')
+    def handle_automation_intent(self, message):
+        LOGGER.debug("Automation trigger intent on entity: "+message.data.get("entity"))
+        message.data["Entity"] = message.data.get("entity")
+        self._handle_automation(message)
+
+    @intent_handler('tracker.intent')
+    def handle_tracker_intent(self, message):
+        LOGGER.debug("Turn on intent on entity: "+message.data.get("entity"))
+        message.data["Entity"] = message.data.get("entity")
+        self._handle_tracker(message)
+
+    @intent_handler('set.climate.intent')
+    def handle_set_thermostat_intent(self, message):
+        LOGGER.debug("Set thermostat intent on entity: "+message.data.get("entity"))
+        message.data["Entity"] = message.data.get("entity")
+        message.data["Temp"] = message.data.get("temp")
+        self._handle_set_thermostat(message)
+
+    @intent_file_handler('add.item.shopping.list.intent')
+    def handle_shopping_list_intent(self, message):
+        LOGGER.debug("Add : "+message.data.get("entity")+"to the shoping list")
```
See PR #51 to be consistent let's clean up these LOGGER statements

stratus-ss commented on __init__.py:
```diff
@@ -192,6 +177,31 @@ def handle_light_decrease_intent(self, message):
+        self._handle_shopping_list(message)
+
     def _handle_switch(self, message):
         LOGGER.debug("Starting Switch Intent")
```
same see PR #51

stratus-ss commented on __init__.py:
```diff
@@ -349,7 +358,7 @@ def _handle_light_adjust(self, message):
-    def handle_automation_intent(self, message):
+    def _handle_automation(self, message):
         entity = message.data["Entity"]
         LOGGER.debug("Entity: %s" % entity)
```
same see PR #51

stratus-ss commented on __init__.py:
```diff
@@ -437,7 +446,7 @@ def _handle_sensor(self, message):
-    def handle_tracker_intent(self, message):
+    def _handle_tracker(self, message):
         entity = message.data["Entity"]
         LOGGER.debug("Entity: %s" % entity)
```
same see PR #51

stratus-ss commented on __init__.py:
```diff
@@ -455,8 +464,8 @@ def handle_tracker_intent(self, message):
+
+    def _handle_set_thermostat(self, message):
         entity = message.data["entity"]
         LOGGER.debug("Entity: %s" % entity)
         LOGGER.debug("This is the message data: %s" % message.data)
```
same see PR #51

stratus-ss reviewed (commented):
Let's make the logging consistent with PR #51

Tony763 committed:
Merge remote-tracking branch 'upstream/20.08' into patch-4

Tony763 commented:
Updated with upstream, changed new `loggers` to `self.log`

Commit:
Merge branch 'patch-4' of https://github.com/Tony763/mycroft-homeassistant into patch-4

Tony763 committed:
Merge branch '20.08' into patch-4

Tony763 commented:
Solved merge conflicts.
@stratus-ss Any progress?

stratus-ss commented:
@Tony763 there are some merge conflicts now because of an update to the translations. Sorry for delaying this one

Tony763 commented:
No problem, I will check it tonight.

Commit:
Updated with upstream

Tony763 commented:
@stratus-ss: Updated, but please try it, my Mycroft went down after Ubuntu upgrade so I cannot test it right now. Hopefully I will manage to make it working in next 2-3 days so I will test it as well.

antipiot committed:
FR-FR translations

Update set.climate.intent

Create homeassistant.shopping.list.dialog

homeassistant.scene.on.dialog

turn.on.intent

turn.off.intent

tracker.intent

Update tracker.intent

toggle.intent

Create sensor.intent

increase.light.brightness.intent

Please check if i've understood the format right for things like "rends (l'|le|la) {{entity}} plus (claire|lumineux|lumineuse)"

decrease.light.brightness.intent

add.item.shopping.list.intent

Create automation.intent

Tony763 commented:
Added Fr-Fr Padatious intents, thanks goes to @antipiot

Tony763 commented:
@stratus-ss:  Hi, any progress?

Tony763 committed:
Changed intent_file_handler to intent_handler

Tony763 commented:
Address also issue #66.

stratus-ss commented:
@Tony763 thanks for bumping this. I had lost track of it. I am going to go back over this on my Mark II and see how it goes

Tony763 commented:
@stratus-ss No problem, let me know, if I should rebase this to something understandable.  After these PRs are done, I will start my work on Issue #64. I have already some vision on mind.

stratus-ss commented:
@Tony763 I am just poking at the automated testing. 

Have you ran the testrunner on your own?

```
mycroft-skill-testrunner /opt/mycroft/skills/homeassistant.mycroftai/
```

For reference it runs and everything passes. I am just trying to spread awareness

stratus-ss commented:
I think that it would be a good time to add some tests for the methods that are being changed.

I know that you are rewriting them, but the original author(s) didn't have tests that I can see for at least some of these. Let's use this opportunity to tighten that up.

Feel free to hit up @krisgesling if you need more help with the tests in this repo. Between the two of us we can probably help to reform this

Tony763 commented:
Hi @stratus-ss, I know about tests. If possible I do not want to add more things to this PR if its working. I do not want to end up in same state as with issue #38.

For tests I had prepared another branch. It contains all tests and also travis based build test with usage of home assistant instance running in each build, so you test thing against real static environment with out put to alluretest. It should be also portable to google actions. Link to [branch](https://github.com/Tony763/mycroft-homeassistant/tree/travis-test)   [Travis](https://travis-ci.org/github/Tony763/mycroft-homeassistant/builds/724152734)   [Allure](https://tony763.github.io/mycroft-homeassistant/8d602f73b1c7d071656af775540a1fac4584ad62/allure_report/)

Please, let me know if this is OK with You and I will prepare PR.

stratus-ss commented:
> Hi @stratus-ss, I know about tests. If possible I do not want to add more things to this PR if its working. I do not want to end up in same state as with issue #38.
> 
> For tests I had prepared another branch. It contains all tests and also travis based build test with usage of home assistant instance running in each build, so you test thing against real static environment with out put to alluretest. It should be also portable to google actions. Link to [branch](https://github.com/Tony763/mycroft-homeassistant/tree/travis-test) [Travis](https://travis-ci.org/github/Tony763/mycroft-homeassistant/builds/724152734) [Allure](https://tony763.github.io/mycroft-homeassistant/8d602f73b1c7d071656af775540a1fac4584ad62/allure_report/)
> 
> Please, let me know if this is OK with You and I will prepare PR.

I'll speak with @krisgesling about this but I think because this is such an important shift tests should be included in order to verify the changes

krisgesling commented on __init__.py:
```diff
@@ -208,6 +193,31 @@ def handle_light_decrease_intent(self, message):
+    @intent_handler('automation.intent')
+    def handle_automation_intent(self, message):
+        self.log.debug("Automation trigger intent on entity: "+message.data.get("entity"))
+        message.data["Entity"] = message.data.get("entity")
```
Is there a reason to reassign these to a capitalized key instead of using the lowercase key in the handler functions?

Tony763 replied:
Yes, reassign was introduced in one of previous PR by one of my predecessor as workaround to prevent loosing data from Padatious core as I remember, maybe I can find a first commit with it.

krisgesling replied:
I'm not really a fan of this, I think it's a good candidate for a follow up refactor but not going to block merging on it.

krisgesling commented on __init__.py:
```diff
@@ -413,12 +422,12 @@ def handle_automation_intent(self, message):
             self.speak_dialog('homeassistant.automation.trigger',
                               data={"dev_name": ha_entity['dev_name']})
-            self.ha.execute_service("homeassistant", "turn_on",
+            self.ha.execute_service("script", "turn_on",
```
Was this a change in the HA API or a bug in the Skill?

Tony763 replied:
Not sure when it was introduced in API but I just wanted to use correct service. You can probably use `homeassistant` for everything, but it's not nice, when there is service specially for it.

Tony763 replied:
It's also same for `scene` in line 430.

krisgesling commented on __init__.py:
```diff
@@ -98,17 +94,6 @@ def on_websettings_changed(self):
         # Otherwise new settings will not be regarded
         self._force_setup()
 
-    def __build_automation_intent(self):
-        intent = IntentBuilder("AutomationIntent").require(
-            "AutomationActionKeyword").require("Entity").build()
-        self.register_intent(intent, self.handle_automation_intent)
-
-    def __build_tracker_intent(self):
-        intent = IntentBuilder("TrackerIntent").require(
-            "DeviceTrackerKeyword").require("Entity").build()
-        # TODO - Identity location, proximity
-        self.register_intent(intent, self.handle_tracker_intent)
-
```
Removing these will break the functionality for any language that hasn't added an `automation.intent` and `tracker.intent`. Could we leave the Adapt intents in with a deprecation warning to give the other languages time to do the translations?

This will also be a reminder to remove all the deprecated voc files eg `AutomationActionKeyword.voc`

Tony763 replied:
I wan to do cleanup in next PR.

For other languages, HA skill is already broken as other intents are already missing. For example, with SV-SE you cannot use even `turn on` as it already miss intens and contain only vocs for it.

I don't know what will be the lesser evil, if adding deprecate message and keeping a bloat in code in already broken language support or just pushing it and wait for users reaction and adding languages on the fly.
I once asked for help with translations for HA plugin to make this process synchronized at forum in past, but did not get  support for all languages.

Languages from active issuers here have support added, namely EN, CS, DE

What I miss are:
FR: `vocab/all_lights.voc` and `vocab/all_switches.voc`  @antipiot 
IT: `vocab/all_lights.voc`,  `vocab/all_switches.voc`, `vocab/tracker.intent` and `vocab/automation.intent `   @kamir86 or @forslund 

Could You please help with translations? Thank you guys in advance

kamir86 replied:
Hi Tony i'll check this out this week

Tony763 replied:
@antipiot, would You help with translation, please? Its only few words.

kamir86 replied:
hi tony, sorry for the late response, I made some PR on your branch with the required translations.
Let me know if I missed something

Tony763 replied:
Hi kamir, Thanks for IT translations, I had seen them when You sent them :) IT seems complete to me, so now I need only FR translations and it should be all. Other languages are already broken from previous commits so to me there is no point hold this PR any longer after FR is completed.

antipiot replied:
> 
> 
> @antipiot, would You help with translation, please? Its only few words.

My apologises, not sure to understand as
 mycroft-homeassistant/vocab/fr-fr/all_switches.voc 
 and 
  mycroft-homeassistant/vocab/fr-fr/all_switches..voc 
Are already translated?

How can i help?

Tony763 replied:
@antipiot, Sorry mate, I see them now, I just did not port them from older branch (turn_on_all) to this one (patch-4). My bad. Doing it right now.

Tony763 replied:
@krisgesling @stratus-ss Hi, working localizations updated, commits squashed, PR should be ready.

Tony763 replied:
@krisgesling @stratus-ss Hi, any updates?

krisgesling replied:
Hey sorry, Stratus has been off for a bit. 

There's no easy way to do these big changes without breaking it for someone. So I'm going to approve this and hopefully we catch anything by people testing the default branch of the repo before it gets updated in the Marketplace.

krisgesling reviewed (commented):
Overall it looks good - however I haven't run it to test the functionality. A few small questions below too.

Regarding the need for tests - I had a quick glance at your work on a new test suite too which looks like it's coming along great! 
Given these aren't in place yet I can see what you mean that adding it would be beyond the scope of this PR. 

I'm going to suggest that we accept this PR based on a manual verification that the intents are triggering as expected, however I do think that getting the full test suite in as soon as possible will significantly improve our rate of development for this Skill as well as the overall reliability of it. 

Once there is a good test process in place we really want each PR to come with it's own tests. Then you should be able to easily see that the tests fail on whatever the current version is, and pass once the PR is fully applied.

Commit:
it-it translations

Added transations by @kamir86

Create homeassistant.scene.on.dialog

Italian translation

Create all_switches.voc

Create all_lights.voc

Create tracker.intent

Update set.light.brightness.intent

Update set.climate.intent

Create automation.intent

Italian translation

Tony763 committed:
FR-FR languages

Added missing files from turn_all-sqash branch, translated by @antipilot
